controllers/semeru_compiler.go

```diff
@@ -33,28 +42,71 @@ const (
 
 // Create the Deployment and Service objects for a Semeru Compiler used by a Websphere Liberty Application
 func (r *ReconcileWebSphereLiberty) reconcileSemeruCompiler(wlva *wlv1.WebSphereLibertyApplication) (error, string) {
+
 	compilerMeta := metav1.ObjectMeta{
 		Name:      wlva.GetName() + SemeruLabelNameSuffix,
 		Namespace: wlva.GetNamespace(),
 	}
-	// Create the Semeru Deployment object
-	semeruDeployment := &appsv1.Deployment{ObjectMeta: compilerMeta}
-	err := r.CreateOrUpdate(semeruDeployment, wlva, func() error {
-		r.reconcileSemeruDeployment(wlva, semeruDeployment)
-		return nil
-	})
-	if err != nil {
-		return err, "Failed to reconcile Deployment : " + semeruDeployment.Name
+
+	semeruCloudCompiler := wlva.GetSemeruCloudCompiler()
+	if semeruCloudCompiler == nil || semeruCloudCompiler.Enabled == nil || !*semeruCloudCompiler.Enabled {
```

Comment:
I like that the cleanup is here, so the controller code is simplified. Please remove the `else` block in controller ([here](https://github.com/WASdev/websphere-liberty-operator/blob/fe397feb97dcc05ac1639c99575bfa6abc8f1ceb/controllers/webspherelibertyapplication_controller.go#L279)) and the if-check above it